libbeat/autodiscover/providers/docker/config.go

```diff
@@ -0,0 +1,18 @@
+package docker
+
+import (
+	"github.com/elastic/beats/libbeat/autodiscover/common"
+	"github.com/elastic/beats/libbeat/processors/add_docker_metadata"
```

Comment:
Feels strange to include here a processor. Perhaps we should extract the TLS config part to a docker package.